ec/internal/gen/serverless/serverless-project-api-dereferenced.yml

```diff
@@ -912,6 +915,13 @@ components:
           $ref: '#/components/schemas/ProjectAlias'
         region_id:
           $ref: '#/components/schemas/RegionID'
+        product_tier:
```

Comment:
The published spec defines [`ObservabilityProjectProductTier`](https://github.com/elastic/serverless-api-specification/blob/main/specifications/generated/production/public-user-serverless-api-dereferenced.yml#L1472-L1478) as a named type. Aside from just keeping things consistent having a named type will mean the generated client doesn't create a different type for the `product_tier` in the create/read/update bodies (e.g https://github.com/elastic/terraform-provider-ec/pull/941/files#diff-1e2497cabe820bf6a8ec4f808164e52c2d0a5eaa70a498a8f26d6361a6a7d311R111). 

Can we update the spec here to define the named component and then just reference that in these three components.